gui/middleware/form.py

```diff
@@ -82,7 +82,7 @@ def __middleware_prepare_value(self, k, v):
         except FieldDoesNotExist:
             pass
         else:
-            if isinstance(field, (DictField, ListField)):
+            if isinstance(field, (DictField, ListField)) and isinstance(v, (str, bytes, bytearray)):
```

Comment:
This was done because in setting specific case, the value of `v` is already a list. So this has been done to only use `json.loads` when `v` is `str`, `bytes` or `bytearray`

Reply:
I get that. The question still stands. Why would we not have list or a dict at this stage? This seems to be working around another bug.